pyccel/ast/numpyext.py

```diff
@@ -786,22 +786,32 @@ def __new__(cls, a, dtype=None, order='K', subok=True, shape=None):
 class NumpyNorm(PyccelInternalFunction):
     """ Represents call to numpy.norm"""
     _dtype = NativeReal()
+    _shape = ()
+    _rank  = 0
 
     def __init__(self, arg, dim=None):
-        if isinstance(dim, ValuedArgument):
-            dim = dim.value
+        super().__init__(arg, dim)
+        cmplx_precision = None
+        if isinstance(arg.dtype, NativeBool):
+            arg = PythonFloat(arg)
+        elif isinstance(arg.dtype, NativeComplex):
+            cmplx_precision = arg.precision
+            arg = PythonAbs(arg)
```

Comment:
I think the addition of the `PythonAbs` should be done in `fcode.py` as it is only a problem in fortran